File: src/Collections/map.ts


import { Wrapped, Thunk, From } from '../../types.ts'

export function New<K, V>(key: Wrapped<K>, val: Wrapped<V>, size: Wrapped<number>): Thunk<Map<K, V>> {
  return () => {
    const data = new Map<K, V>()

    const tgt:number = From(size)
    for (let idx = 0; idx < tgt; idx++) {
      data.set(From(key), From(val))
    }

    return data
  }
}

File: collection/map/map.go
package kvmap

import (
	. "github.com/rgrannell1/peach/types"
)

/*
 * Construct an array of values
 */
func New[K comparable, V any](key Wrapped[K], val Wrapped[V], size Wrapped[int]) Wrapped[map[K]V] {
	return Thunk(func () map[K]V {
		data := map[K]V{}
		tgt := size.From()

		for idx := 0; idx < tgt; idx++ {
			data[key.From()] = val.From()
		}

		return data
	})
}
